sdk/cosmos/azure-cosmos-benchmark/src/main/java/com/azure/cosmos/benchmark/SyncBenchmark.java

```diff
@@ -0,0 +1,359 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.benchmark;
+
+import com.azure.cosmos.BridgeInternal;
+import com.azure.cosmos.CosmosClient;
+import com.azure.cosmos.CosmosClientBuilder;
+import com.azure.cosmos.CosmosContainer;
+import com.azure.cosmos.CosmosItemResponse;
+import com.azure.cosmos.implementation.Utils;
+import com.codahale.metrics.ConsoleReporter;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.ScheduledReporter;
+import com.codahale.metrics.Timer;
+import com.codahale.metrics.graphite.Graphite;
+import com.codahale.metrics.graphite.GraphiteReporter;
+import com.codahale.metrics.jvm.CachedThreadStatesGaugeSet;
+import com.codahale.metrics.jvm.GarbageCollectorMetricSet;
+import com.codahale.metrics.jvm.MemoryUsageGaugeSet;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetSocketAddress;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+abstract class SyncBenchmark<T> {
+    private final MetricRegistry metricsRegistry = new MetricRegistry();
+    private final ScheduledReporter reporter;
+    private final ExecutorService executorService;
+
+    private Meter successMeter;
+    private Meter failureMeter;
+
+    final Logger logger;
+    final CosmosClient cosmosClient;
+    final CosmosContainer cosmosContainer;
+
+    final String partitionKey;
+    final Configuration configuration;
+    final List<PojoizedJson> docsToRead;
+    final Semaphore concurrencyControlSemaphore;
+    Timer latency;
+
+    abstract class ResultHandler<T, Throwable> implements BiFunction<T, Throwable, T> {
+        ResultHandler() {
+        }
+
+        protected void init() {
+        }
+
+        @Override
+        abstract public T apply(T o, Throwable throwable);
+    }
+
+    class LatencyListener<T> extends ResultHandler<T, Throwable> {
```

Comment:
This is a good example of nested class (non-static) which uses enclosing class's members (`latency` here)